File: plugin/src/test/java/io/github/pshevche/act/ActPluginFuncTest.java
package io.github.pshevche.act;

import org.gradle.testkit.runner.BuildResult;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertTrue;

class ActPluginFuncTest extends BaseActPluginFuncTest {

    @Test
    void canRunTask() {
        BuildResult result = run("greeting");

        // Verify the result
        assertTrue(result.getOutput().contains("Hello from plugin 'io.github.pshevche.greeting'"));
    }
}

File: plugin/src/test/kotlin/io/github/pshevche/act/ActPluginFuncTest.kt
package io.github.pshevche.act

import io.github.pshevche.act.extensions.BuildRunner
import io.github.pshevche.act.extensions.Workspace
import io.kotest.core.spec.style.FreeSpec
import io.kotest.engine.spec.tempdir
import io.kotest.matchers.string.shouldContain

class ActPluginFuncTest : FreeSpec({

    val workspace = extension(Workspace(tempdir()))
    val runner = BuildRunner(workspace)

    "can run task" {
        val result = runner.build("greeting")
        result.output shouldContain "Hello from plugin 'io.github.pshevche.act'"
    }
})
